runtime/integration-tests/src/connectors_gateway/gateway.rs

```diff
@@ -0,0 +1,270 @@
+// Copyright 2023 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+use cfg_primitives::{Balance, PoolId, TrancheId};
+use cfg_traits::connectors::Codec;
+use cfg_types::{
+	domain_address::{Domain, DomainAddress},
+	fixed_point::Rate,
+	tokens::{CurrencyId, CustomMetadata},
+};
+use connectors_gateway_routers::{
+	axelar_evm::AxelarEVMRouter, ethereum_xcm::EthereumXCMRouter, DomainRouter, EVMChain,
+	EVMDomain, FeeValues, XCMRouter, XcmDomain, XcmTransactInfo,
+};
+use development_runtime::xcm::CurrencyIdConvert;
+use frame_support::{
+	assert_noop, assert_ok,
+	dispatch::{GetDispatchInfo, Pays},
+	weights::Weight,
+};
+use fudge::primitives::Chain;
+use orml_traits::asset_registry::AssetMetadata;
+use pallet_connectors::Message;
+use pallet_connectors_gateway::GatewayOrigin;
+use pallet_democracy::{AccountVote, Conviction, ReferendumIndex, Vote, VoteThreshold};
+use sp_core::{bounded::BoundedVec, bounded_vec, H160, H256};
+use sp_runtime::{
+	traits::{BlakeTwo256, Convert, Hash},
+	Storage,
+};
+use tokio::runtime::Handle;
+use xcm::{
+	latest::{Junction, Junctions, MultiLocation},
+	VersionedMultiLocation,
+};
+
+use crate::{
+	chain::centrifuge::{
+		AccountId, CouncilCollective, FastTrackVotingPeriod, MinimumDeposit, Runtime, RuntimeCall,
+		RuntimeEvent, PARA_ID,
+	},
+	utils::{
+		accounts::Keyring,
+		collective::{collective_close, collective_propose, collective_vote},
+		connectors_gateway::{add_connector, remove_connector, set_domain_router},
+		democracy::{democracy_vote, execute_via_democracy, external_propose_majority, fast_track},
+		env::{ChainState, EventRange},
+		preimage::note_preimage,
+		*,
+	},
+};
+
+fn get_council_members() -> Vec<Keyring> {
+	vec![Keyring::Alice, Keyring::Bob, Keyring::Charlie]
+}
+
+#[tokio::test]
+async fn set_router() {
```

Comment:
This is something that @mustermeiszer wanted to have for these tests so we can test the whole flow. I also agree with the idea that this might be out of scope, however, we could also argue that we are confirming that the pallet is set up correctly by working with governance.